src/Adornment/Adornment.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ReactNode, cloneElement, RefObject } from 'react'
+
+import { AdornmentWrapper, StyledAdornment } from './styled'
+import { useRefDependentCssLengthValue } from '../useRefDependentCssLengthValue'
+import { isReactElement } from '../utils/isReactElement'
+
+export enum Position {
+  LEFT = 'Left',
+  RIGHT = 'Right'
+}
+
+type AdornmentProps = {
+  children: ReactNode
+
+  leftComponent?: ReactNode
+  rightComponent?: ReactNode
+  inset?: number
+}
+
+function tryGetRectWidth (ref: RefObject<HTMLElement>) {
```

Comment:
I dont really like this - if i need a function like this i would rather use something like:
```
getNodeWidth = (node) => node.getBoundingClientRect().width
```

But for this usecase I would use the following advised pattern:
```
function useClientRect() {
  const [rect, setRect] = useState(null)

  const ref = useCallback((node) => {
    if (node !== null) {
      setRect(node.getBoundingClientRect())
    }
  }, [])

  return [rect, ref]
}
```